Troubleshooting.md

```diff
@@ -404,6 +406,32 @@ If this error occurs within a virtual environment, try reinstalling the libs by
 1. Install bz2 headers. Use the command `sudo apt-get install libbz2-dev` on Ubuntu.
 2. Install Python 3 again so that the bz2 library gets included in `~/.pyenv/versions/3.8.15/lib/python3.8/`. Use the command `pyenv install 3.8.15`.
 
+### Subprocess.CalledProcessError: Command 'yarn install --pure-lockfile' returned non-zero exit status 1
+
+If you get error like this when running ``python -m scripts.start``:
+```
+Traceback (most recent call last):
```

Comment:
This is the traceback, can you add a few lines of preceding error logs as well, maybe around 6-7 lines. 

Because if you see these discussions https://github.com/oppia/oppia/discussions/17425, https://github.com/oppia/oppia/discussions/18428, https://github.com/oppia/oppia/discussions/14814, these all have this same error but different reasons to why it occurred and thus different solutions.

Reply:
@Ash-2k3 The error log I pasted here is from this [comment](https://github.com/oppia/oppia/discussions/16983#discussioncomment-8610303). Can you please tell me if there is a way to reproduce these errors? I am not sure about the exact error logs I encountered as it has been a month and I can't recall them precisely.

Reply:
I am not sure what to do in this case, maybe apart from your changes, let's provide links to these issues that I mentioned and ask them to follow those, if your solution didn't work for them ? wdyt

Reply:
@Ash-2k3 After my solution ends I will put a note telling that this is very generalized solution, if this solution doesn't work well with folks so they can look the issues listed below!! how about this?

Reply:
Sounds good, thanks!

Reply:
Done!